### PR TITLE
Support sync bucket in lightning

### LINKED ISSUE #1627: Support sync bucket in lightning
In tpcc test, we may place the csv files in bucket directly, it would great if lightning could support such data source.

### PR DESCRIPTION
**Pull request by aylei**
Signed-off-by: Aylei <rayingecho@gmail.com>

close #1627 

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

### Does this PR introduce a user-facing change?:

```release-note
support using a remote directory as data source for tidb-lightning.
```

### CODE REVIEW

File: charts/tidb-lightning/values.yaml [hunk 1]
```diff
@@ -30,6 +30,8 @@ dataSource:
     storage: 100Gi
     secretName: cloud-storage-secret
     path: s3:bench-data-us/sysbench/sbtest_16_1e7.tar.gz
+    # Directory support downloading all files in a remote directory, shadow dataSoure.remote.path if present
```
Comment (tennix) [inline charts/tidb-lightning/values.yaml]: What about reusing the `path` for this? If a path ends with `/`, then it is a directory.

Comment (Yisaer) [reply charts/tidb-lightning/values.yaml]: I agree with the current solution, reusing the `path` might be more complex for developers and users.

**Comment by cofyc** [comment]
resolve conflicted files?

**Comment by aylei** [comment]
@cofyc @tennix @Yisaer I've fixed a merge conflict, PTAL again

**Comment by Yisaer** [comment]
@aylei  plz cherry-pick this request to release-1.1 manually.

**Pull request merged**